### PR TITLE
DM: update openapi spec

### PR DESCRIPTION
**Pull request by Ehco1996**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/8612
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-open-api.md [hunk 1]
```diff
@@ -248,13 +268,218 @@ curl -X 'POST' \
...
+  'http://127.0.0.1:8261/api/v1/sources/mysql-01?force=true' \
+  -H 'accept: application/json'
+```
+
+## Update a data source
+
+This API is a synchronous interface. If the request is successful, the information of the corresponding data source is returned.
+> **NOTE:**
+>
+> Updating the data source configuration requires that there are no running tasks under the current data source
```
Comment (TomShawn) [inline dm/dm-open-api.md:362]:
```suggestion

> **NOTE:**
>
> When you use this API to update the data source configuration, make sure that there are no running tasks under the current data source.
```

File: dm/dm-open-api.md [hunk 2]
```diff
@@ -248,13 +268,218 @@ curl -X 'POST' \
...
+  },
+  "purge": {
+    "interval": 3600,
+    "expires": 0,
+    "remain_space": 15
+  }
+}
+```
+
+## Enable-a data source
```
Comment (TomShawn) [inline dm/dm-open-api.md:438]:
```suggestion
## Enable a data source
```

File: dm/dm-open-api.md [hunk 3]
```diff
@@ -248,13 +268,218 @@ curl -X 'POST' \
...
+
+This is a synchronisation interface that enables this data source on a successful request and starts all subtasks of the task that rely on this data source in bulk.
```
Comment (TomShawn) [inline dm/dm-open-api.md:440]:
```suggestion
This is a synchronous interface that enables a data source on a successful request and starts all subtasks of the task that rely on this data source in batch.
```

File: dm/dm-open-api.md [hunk 4]
```diff
@@ -248,13 +268,218 @@ curl -X 'POST' \
...
+```shell
+curl -X 'POST' \
+  'http://127.0.0.1:8261/api/v1/sources/mysql-01/enable' \
+  -H 'accept: */*' \
+  -H 'Content-Type: application/json'
+```
+
+## Disable a data source
+
+This is a synchronisation interface that deactivates this data source on a successful request and stops all subtasks of the task that depend on it in bulk.
```
Comment (TomShawn) [inline dm/dm-open-api.md:459]:
```suggestion
This is a synchronous interface that deactivates this data source on a successful request and stops all subtasks of the task that rely on it in batch.
```

File: dm/dm-open-api.md [hunk 5]
```diff
@@ -410,40 +617,27 @@ curl -X 'POST' \
...
-{{< copyable "shell-regular" >}}
-
-```shell
-curl -X 'POST' \
-  'http://127.0.0.1:8261/api/v1/sources/mysql-01/pause-relay' \
-  -H 'accept: */*'
-```
-
-## Resume the relay-log feature for data sources
+## Purge relay log files that no longer required
```
Comment (TomShawn) [inline dm/dm-open-api.md:620]:
```suggestion
## Purge relay log files that are no longer required
```

File: dm/dm-open-api.md [hunk 6]
```diff
@@ -518,11 +712,11 @@ curl -X 'GET' \
 
 ## Create a replication task
 
-This API is an asynchronous interface. If the request is successful, the status code of the returned body is 200. To learn about its latest status, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
+This API is an synchronous interface. If the request is successful, the status code of the returned body is 200. a successful request will return information about the corresponding synchronisation task.
```
Comment (TomShawn) [inline dm/dm-open-api.md:715]:
```suggestion
This API is a synchronous interface. If the request is successful, the status code of the returned body is 200. A successful request will return the information of the corresponding replication task.
```

File: dm/dm-open-api.md [hunk 7]
```diff
@@ -716,133 +906,364 @@ curl -X 'POST' \
...
+
+```shell
+curl -X 'DELETE' \
+  'http://127.0.0.1:8261/api/v1/tasks/task-1' \
+  -H 'accept: application/json'
+```
+
+## Update a replication task
+
+This interface is a synchronisation interface and a successful request will return information about the task.
```
Comment (TomShawn) [inline dm/dm-open-api.md:1036]:
```suggestion
This interface is a synchronous interface and a successful request returns the information of the task.
```

File: dm/dm-open-api.md [hunk 8]
```diff
@@ -716,133 +906,364 @@ curl -X 'POST' \
...
+
+> **NOTE:**
+>
+> Updating the task configuration requires that the task is stopped and has run into incremental sync and that only some of the fields can be updated.
```
Comment (TomShawn) [inline dm/dm-open-api.md:1040]:
```suggestion
> **Note:**
>
> When you use this API to update the task configuration, make sure that the task is stopped and has run into incremental sync and that only some of the fields can be updated.
```

File: dm/dm-open-api.md [hunk 9]
```diff
@@ -716,133 +906,364 @@ curl -X 'POST' \
...
+        "binlog_gtid": "03fc0263-28c7-11e7-a653-6c0b84d59f30:1-7041423,05474d3c-28c7-11e7-8352-203db246dd3d:1-170"
+      }
+    ]
+  }
 }
 ```
 
+## Start a replication task
+
+This API is an asynchronous interface. If the request is successful, the status code of the returned body is 204. To learn about its latest status, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
```
Comment (TomShawn) [inline dm/dm-open-api.md:1236]:
```suggestion
This API is an asynchronous interface. If the request is successful, the status code of the returned body is 204. To learn the latest status of a task, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
```

File: dm/dm-open-api.md [hunk 10]
```diff
@@ -716,133 +906,364 @@ curl -X 'POST' \
...
+
+```shell
+curl -X 'POST' \
+  'http://127.0.0.1:8261/api/v1/tasks/task-1/start' \
+  -H 'accept: */*'
+```
+
 ## Stop a replication task
 
 This API is an asynchronous interface. If the request is successful, the status code of the returned body is 204. To learn about its latest status, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
```
Comment (TomShawn) [inline dm/dm-open-api.md:1254]:
```suggestion
This API is an asynchronous interface. If the request is successful, the status code of the returned body is 200. To learn the latest status of a task, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
```

File: dm/dm-open-api.md [hunk 11]
```diff
@@ -917,48 +1338,148 @@ curl -X 'GET' \
 }
 ```
 
-## Pause a replication task
+## Get the replication task list
 
-This API is an asynchronous interface. If the request is successful, the status code of the returned body is 200. To learn about its latest status, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
+This API is a synchronous interface. If the request is successful, the information of the corresponding node is returned.
```
Comment (TomShawn) [inline dm/dm-open-api.md:1343]: 这里中文是【请求成功会返回对应的同步任务信息】，似乎不太对应呢：https://github.com/pingcap/docs-cn/pull/8612/files#diff-2d25f555740c97a16d90319fb4179091a04117e4ab95354d871cf2712c04e85eR1344

File: dm/dm-open-api.md [hunk 12]
```diff
@@ -917,48 +1338,148 @@ curl -X 'GET' \
...
+            "binlog_gtid": "03fc0263-28c7-11e7-a653-6c0b84d59f30:1-7041423,05474d3c-28c7-11e7-8352-203db246dd3d:1-170"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+## Get the replication task migrate rules
```
Comment (TomShawn) [inline dm/dm-open-api.md:1453]:
```suggestion
## Get the migration rules of a replication task
```

File: dm/dm-open-api.md [hunk 13]
```diff
@@ -917,48 +1338,148 @@ curl -X 'GET' \
...
 
-This API is an asynchronous interface. If the request is successful, the status code of the returned body is 200. To learn about its latest status, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
+This API is a synchronous interface and a successful request will return the corresponding list.
```
Comment (TomShawn) [inline dm/dm-open-api.md:1455]:
```suggestion
This API is a synchronous interface and a successful request returns a list of the corresponding tasks.
```

Comment (TomShawn) [inline dm/dm-open-api.md:1455]:
```suggestion
This API is a synchronous interface and a successful request returns a list of the corresponding tasks.
```

File: dm/dm-open-api.md (hunk 11 above)
Comment (Ehco1996) [reply dm/dm-open-api.md]: 这里之前的中文描述就不太准确，感谢指正，在这里修复了

https://github.com/pingcap/docs-cn/pull/8612/commits/eaac731c2700aaff6c007a69ede874291624d071
https://github.com/pingcap/docs/pull/7866/commits/2da947d5f1d1f9dd0a80902aaf278be118b5e8ef

File: dm/dm-open-api.md [hunk 14]
```diff
@@ -5,11 +5,23 @@ summary: Learn about how to use OpenAPI interface to manage the cluster status a
...
-```
+To enable OpenAPI, perform one of the following operations:
+
++ If your DM cluster has been deployed directly using binary, add the following configuration to the DM-master configuration file.
+
+    ```toml
+    openapi = true
+    ```
+
++ If your DM cluster is deployed using TiUP, add the following configuration to the topology file：
```
Comment (TomShawn) [inline dm/dm-open-api.md:18]:
```suggestion
+ If your DM cluster has been deployed using TiUP, add the following configuration to the topology file：
```

File: dm/dm-open-api.md [hunk 15]
```diff
@@ -917,48 +1339,148 @@ curl -X 'GET' \
...
+      }
+    }
+  ]
+}
+```
+
+## Get the migration rules of a replication task
 
-This API is an asynchronous interface. If the request is successful, the status code of the returned body is 200. To learn about its latest status, You can [get the information of a replication task](#get-the-information-of-a-replication-task).
+This API is a synchronous interface and a successful request returns a list of the migrate rules of this task.
```
Comment (TomShawn) [inline dm/dm-open-api.md:1456]:
```suggestion
This API is a synchronous interface and a successful request returns a list of the migration rules of this task.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**